src/main/java/com/listywave/user/application/domain/User.java

```diff
@@ -107,13 +110,20 @@ public void validateUpdate(Long id) {
     }
 
     public void follow(User followingUser) {
+        if (this.followingCount == MAX_FOLLOW_COUNT) {
+            throw new CustomException(EXCEED_FOLLOW_COUNT_EXCEPTION, "팔로우는 최대 " + MAX_FOLLOW_COUNT + "명까지 가능합니다.");
+        }
         this.followingCount++;
         followingUser.followerCount++;
     }
 
     public void unfollow(User followingUser) {
-        this.followingCount--;
-        followingUser.followerCount--;
+        if (this.followingCount > 0) {
+            this.followingCount--;
+        }
+        if (followingUser.followerCount > 0) {
+            followingUser.followerCount--;
+        }
```

Comment:
메소드는 하나의 기능만 수행하는게 더 나아보입니다!(SRP : 단일책임원칙) 각각 파트를 메소드로 빼놓는게 어떨까요!?
추후 회원 탈퇴때도 각각 감소 부분 메소드를 재사용 할 수 있을 거 같아요

아래 코드를 제안해봅니다!
```suggestion
  public void unfollow(User followingUser) {
    this.decreaseFollowingCount();
    followingUser.decreaseFollowerCount();
}

private void decreaseFollowingCount() {
    if (this.followingCount > 0) {
        this.followingCount--;
    }
}

private void decreaseFollowerCount() {
    if (this.followerCount > 0) {
        this.followerCount--;
    }
}
```

Reply:
곧 구현할 회원 탈퇴 로직에서 재사용될 메리트가 커보이네요! 좋습니다 👍👍👍